packages/framework/src/Component/Router/FriendlyUrl/FriendlyUrlFactory.php

```diff
@@ -95,4 +103,17 @@ public function createForAllDomains(
 
         return $friendlyUrls;
     }
+
+    /**
+     * @param string $routeName
+     * @return bool
+     */
+    protected function isRouteMultidomain(string $routeName): bool
+    {
+        $friendlyUrlRouter = $this->domainRouterFactory->getFriendlyUrlRouter($this->domain->getDomainConfigById(Domain::FIRST_DOMAIN_ID));
+        $routeCollection = $friendlyUrlRouter->getRouteCollection();
+        $route = $routeCollection->get($routeName);
+
+        return $route?->getOption('multidomain') ?? true;
```

Comment:
it would be nice to cover the case when $route is null with some RouteNotFindException as in that case, it will return true, right? But I do not insist on that